Library/Homebrew/formula_auditor.rb

```diff
@@ -1064,12 +1141,57 @@ def linux_only_gcc_dep?(formula)
       true
     end
 
-    def committed_version_info
-      return [] unless @git
-      return [] unless formula.tap # skip formula not from core or any taps
-      return [] unless formula.tap.git? # git log is required
-      return [] if formula.stable.blank?
-      return [@previous_committed, @newest_committed] if @previous_committed.present? || @newest_committed.present?
+    sig { params(tap: Tap, only_names: T.nilable(T::Array[String])).returns(T::Array[Pathname]) }
+    def changed_formulae_paths(tap, only_names: nil)
+      return [] unless tap.git?
+
+      base_ref = "origin/HEAD"
+      changed_paths = Utils.safe_popen_read(Utils::Git.git, "-C", tap.path, "diff", "--name-only", base_ref)
+                           .lines
+                           .filter_map do |line|
+        relative_path = line.chomp
+        next unless relative_path.end_with?(".rb")
+
+        absolute_path = tap.path/relative_path
+        next unless absolute_path.exist?
+        next unless absolute_path.to_s.start_with?(tap.formula_dir.to_s)
+
+        absolute_path
+      end
+
+      return changed_paths if only_names.blank?
+
+      expected_paths = only_names.filter_map do |name|
+        relative_name = name.to_s.delete_prefix("#{tap.name}/")
+        relative_path = Pathname(relative_name)
+        relative_path = relative_path.sub_ext(".rb") if relative_path.extname.empty?
+        absolute_path = tap.formula_dir/relative_path
+        absolute_path.expand_path if absolute_path.exist?
+      rescue
+        nil
+      end.map(&:to_s)
+
+      return [] if expected_paths.empty?
+
+      changed_paths.select { |path| expected_paths.include?(path.expand_path.to_s) }
+    end
+
+    def committed_version_info(formula: @formula)
+      empty_result = [{}, {}]
+      return empty_result unless @git
+      return empty_result unless formula.tap # skip formula not from core or any taps
+      return empty_result unless formula.tap.git? # git log is required
+      return empty_result if formula.stable.blank?
+
+      cache_key = if formula.respond_to?(:full_name)
+        formula.full_name
+      else
+        formula.name
+      end
+      return @committed_version_info_cache[cache_key] if @committed_version_info_cache.key?(cache_key)
+
+      previous_committed = {}
+      newest_committed = {}
```

Comment:
I know this isn't from this PR, but I think this terminology is confusing. Maybe something like this?

```suggestion
      version_info_at_previous_version = {}
      version_info_at_origin_head = {}
```

Reply:
@Rylan12 thanks for nudge. Not used your naming exactly but have riffed off it, been consistent across the whole file and think it's made things much more readable. Thanks for the nudge!